### DISCUSSION TITLE
Switching worlds

### DISCUSSION THREAD

**Discussion by BstAA** [itzg/docker-minecraft-server#811] [Q&A]
Hi Geoff,

first THX for this really impressive server!

A question regarding switching worlds: Your README states that different maps should be placed below the `worlds `folder
```
/path/on/host
├── worlds
│   └── ... PLACE MAPS IN THEIR OWN FOLDERS HERE ...
└── ...
```
and than 

"Once your maps are in the proper path, you can then specify which map the server uses by changing the `level-name` value in `server.properties` to match the name of your map."

If I take this way the selected map is `copied `to the `data` folder and not just used. So finally the subfolder `worlds` only has the "old" worlds.

How can I avoid this?

Klaus

**Comment by itzg** [comment]
I never have liked that directory structure diagram I created there, so your question inspired me to overhaul it into a new section with a (hopefully) better diagram:

https://github.com/itzg/docker-minecraft-server#data-directory

So, to answer your question more broadly, the `level-name`/`LEVEL` setting for the worlds directory is a separate but related concept from the `WORLD` setting.

**Reply by BstAA** [reply]
THX!
So to clarify further. Assume I've got two maps 

```
/path/on/host
├── worlds
│   └── w1
│   └── w2
└── ...
```
and just want to switch between them in creative mode. What do I have to do?

Using [World Save Name](https://github.com/itzg/docker-minecraft-server#world-save-name) i.e. just switching between  `LEVEL=w1` and `LEVEL=w2`?

Using `level-name=w1 `in `server.properties `resulted in 

```
/path/on/host
├── w1 (copied from w1 below)
├── worlds
│   └── w1
│   └── w2
└── ...
```
O should I add a path like `LEVEL=worlds/w2` ?
I don't want to copy, but just use.

By the way I couldn't find `OVERWRITE_SERVER_PROPERTIES ` in the docs.

**Reply by itzg** [reply]
I'm sorry, I typo'ed that variable name, I meant OVERRIDE_SERVER_PROPERTIES from 

https://github.com/itzg/docker-minecraft-server#server-configuration

There might be a bug lurking because LEVEL/level-name shouldn't be doing any copying of world directories especially not to the top level of the data path. Can you provide your full container configuration?

**Reply by BstAA** [reply]
This is my test case:
Start with the following `docker-compose.yml`:


```
version: "3"

services:
  mc-test:
    image: itzg/minecraft-server
# Test only
#    ports:
#      - 25565:25565
    environment:
      EULA: "TRUE"
      GID: "100"
      UID: "1000"
    volumes:
      - ./data:/data
      - /etc/timezone:/etc/timezone:ro
    restart: unless-stopped

```
Result:

```
.../mc-test/data$ tree -L 1
.
├── minecraft_server.1.16.5.jar
├── server.properties
├── ... 
└── world
```
stop service; create `worlds`  directory; duplicate  `world` to `w1` and `w2`; edit server.properties

```
.../mc-test/data$ tree -L 3
.
├── server.properties
└── worlds
    ├── w1
    │   ├── DIM-1
    │   ├── ...
    └── w2
        ├── DIM-1
        ├── ...
```

```
$ cat server.properties | grep level-name
level-name=w1
$ cd ..
$ docker-compose start
$ cd data
.../data$ tree -L 2
.
├── w1
├── ...
└── worlds
    ├── w1
    └── w2
```

Map `w1 `is _copied_ to the `data` directory

**Reply by itzg** [reply]
Well, this is embarrassing/strange. Somewhere along the way I must have confused myself about the directory structure of the Java and Bedrock editions and mixed up the two. I fired up a fresh, minimal container and indeed the Java edition servers place the `$WORLD` directory at the top level and **not** inside a `worlds` directory.

I need to get that diagram and section updated ASAP, because it is incorrect.

**Reply by BstAA** [reply]
Hi Geoff,

will you change the behavior i.e. `$WOLRD ` _inside_ the `worlds ` or just change the
description?

**Reply by itzg** [reply]
Just the description, because the behavior is beyond my control.

**Reply by itzg** [reply]
Diagram has been updated.

**Reply by itzg** [reply]
> THX!
> 
> So to clarify further. Assume I've got two maps 
> 
> 
> 
> ```
> 
> /path/on/host
> 
> ├── worlds
> 
> │   └── w1
> 
> │   └── w2
> 
> └── ...
> 
> ```
> 
> and just want to switch between them in creative mode. What do I have to do?
> 
> 
> 
> Using [World Save Name](https://github.com/itzg/docker-minecraft-server#world-save-name) i.e. just switching between  `LEVEL=w1` and `LEVEL=w2`?
> 
> 
> 
> Using `level-name=w1 `in `server.properties `resulted in 
> 
> 
> 
> ```
> 
> /path/on/host
> 
> ├── w1 (copied from w1 below)
> 
> ├── worlds
> 
> │   └── w1
> 
> │   └── w2
> 
> └── ...
> 
> ```
> 
> O should I add a path like `LEVEL=worlds/w2` ?
> 
> I don't want to copy, but just use.
> 
> 
> 
> By the way I couldn't find `OVERWRITE_SERVER_PROPERTIES ` in the docs.

To answer this original request, make sure w1 and w2 are pre-situated in `/data` and then you can switch between them. 

I suspect what was happening before is that the Minecraft server was ignoring the `worlds` and simply regenerating a new world in `/data/w1`.

**Comment by BstAA** [comment]
> To answer this original request, make sure w1 and w2 are pre-situated in /data and then you can switch between them.
That's what I did. 

> I suspect what was happening before is that the Minecraft server was ignoring the worlds and simply regenerating a new world in /data/w1.

No. It's not _regenerating_ a new world, but _copies_ the respective world folder from `worlds `to `/data`.

**Comment by Stuebi5** [comment]
Hello There

How is it possible to switch to a different world now?
Does not seem like it is, how it used to be:
![image](https://user-images.githubusercontent.com/102486517/170833690-370ef4c3-3179-4607-a307-26b708b37ad1.png)
It got 3 Folders now, which splits the world up.

Anyone can help?

**Reply by itzg** [reply]
That how Paper/Spigot organizes each dimension of the world.
